NEWPART.md

```diff
@@ -0,0 +1,203 @@
+
+# Quick guide to adding a new part to an existing family
+
+This documents what I did to add support for xc7a100t parts, 
+in hope that it is useful to someone adding another new part.
+
+_If you are **just** adding a new package for a device that is already handled, you can skip Steps 2..4!_
+
+
+### Step 0 ###
+Fork a copy of https://github.com/SymbiFlow/prjxray on GitHub (go to the page, click "Fork" button, select your own workspace).
+
+Clone your fork (replace "tcal-x" with your GitHub ID):
+
+```
+git clone git@github.com:tcal-x/prjxray.git
+cd prjxray
+```
+
+
+
+### Step 1 ###
+
+Follow the Project X-Ray setup instructions in the [README.md](https://github.com/SymbiFlow/prjxray/blob/master/README.md), up through Step 7, Option 1.  That is, source the `./download-latest-db.sh` script.   NOTE: if you might want to run tests with your own generated db, you'll need to take some extra steps here.   I sketch out what that process looks like at the end of this document.  
+
+
+
+### Step 2 ###
+
+Add a new settings file.  Usually you will start with an existing settings file and modify it.
+Assuming you're in prjxray/,
+
+```
+cp settings/artix7_200t.sh settings/artix7_100t.sh
+git add settings/artix7_100t.sh
+```
+
+Update the following values in the new settings file:
+* `XRAY_PART` -- choose a package that is fully bonded (typically the largest number of pins)
+* `XRAY_ROI_TILEGRID` -- modify the bounding boxes to be a tight fit on your new part.
+* `XRAY_IOI3_TILES` -- See the comments in 005 fuzzers.  Special handling.  Ask Keith or Tim or Tim.
+* `XRAY_PIN_00` -- this must be a clock pin.   You can look at the device in the Vivado GUI interactively (click on an IO and look at its properties until you find one with IS\_CLOCK=true), or  run a small test in Vivado and see which pin assigned the 'clk'.
+* `XRAY_PIN_01` and on -- these should be normal data pins on the device.
+
+Source this new settings file:
+
+```
+source settings/artix7_100t.sh
+```
+
+### Step 3 ###
+
+Edit the top Makefile
+
+* Add the new part to the correct list.   This is the basename of the new settings file that you just created. In this case, we added `artix7_100t`:
+
+```
+ARTIX_PARTS=artix7_200t artix7_100t
+```
+
+
+### Step 4 ###
+
+Make the following target, adapted for your new part, and setting the parallelism factor appropriate for your machine:
+
+```
+make -j32 MAX_VIVADO_PROCESS=32 db-part-only-artix7_100t
+```
+
+* It should run fuzzers 000, 001, 005, 072, 073, 074, and 075.
+                
+* 005 will take a long time unless you throw a lot of cores at it.  32-64 cores suggested.
+
+* You _may_ get a spurious failure, since the parallel runs may violate ordering dependencies among the fuzzers.  If this happens, clean the fuzzer that failed (e.g.  `make -C fuzzers/005-tilegrid clean`), and then rerun the top-level make.
```

Comment:
There shouldn't be ordering failing in 005?

Reply:
001 must produce the parts yaml for 005.    But yeah, now when I run the target `db-part-only-<part>`, I see that 005 waits for 001 to finish.  Do you think that respecting the dependency was added recently, or did I just see some fluke, maybe due to a restart, where 005 started when the parts yaml wasn't there?   If this should never happen, I'll take out the section.

Reply:
I believe there shouldn't be an issue here, and if there was we'd want to file a bug, replicate the issue and then fix it.